src/test/ui/parser-recovery-4.stderr

```diff
@@ -0,0 +1,34 @@
+error: incorrect close delimiter: `}`
+  --> $DIR/parser-recovery-4.rs:20:1
+   |
+LL | fn main() {
+   |           - close delimiter possibly meant for this
+LL |     let x = 1;
+LL |     foo(x
+   |        - un-closed delimiter
+...
+LL | }
+   | ^ incorrect close delimiter
+
+error: expected one of `!`, `)`, `,`, `.`, `::`, `?`, `{`, or an operator, found `bar`
```

Comment:
something feels..inelegant to me here. Like, I would like to be suppressing these "fallout" errors, and just picking up parsing from some other point..? Hmm, I'm not sure.

Reply:
That is an option, to just bail on parsing the entire block from that point on, as we do in other parts of the parser.

Reply:
The problem with out right bailing until the next closing brace is that we then don't get the _nice recovery_ shown above. Maybe that's a reasonable trade-off...

Reply:
which is the nice recovery you are referring to?

I tend to think that giving "false errors" is kind of the most confusing thing we can do and we should bias in favor of too few errors and not too many, but opinions vary.